File: Views/Tutorial/WidgetTutorialView.swift
import SwiftUI

struct WidgetTutorialView: View {
    @Environment(\.presentationMode) var presentationMode
    @State private var currentPage = 0
    
    private let tutorialSteps = [
        TutorialStep(
            title: "Restez appuyer sur votre écran d'accueil puis cliquez sur le +",
            description: "",
            imageName: "imageA"
        ),
        TutorialStep(
            title: "Cliquez sur Ajouter un Widget",
            description: "",
            imageName: "imageB"
        ),
        TutorialStep(
            title: "Cherchez Love2Love",
            description: "",
            imageName: "imageC"
        ),
        TutorialStep(
            title: "Sélectionnez le Widget que vous souhaitez puis cliquer sur ajouter",
            description: "",
            imageName: "imageD"
        )
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            // Handle pour indiquer que c'est une sheet
            RoundedRectangle(cornerRadius: 2.5)
                .fill(Color.gray.opacity(0.3))
                .frame(width: 40, height: 5)
                .padding(.top, 12)
                .padding(.bottom, 8)
            
            // Titre dynamique de l'étape courante
            Text(tutorialSteps[currentPage].title)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.black)
                .multilineTextAlignment(.center)
                .padding(.top, 20)
                .padding(.horizontal, 16)
                .padding(.bottom, 20)
                .lineLimit(nil)
                .fixedSize(horizontal: false, vertical: true)
                .animation(.easeInOut(duration: 0.3), value: currentPage)
            
            // Contenu du tutoriel avec pagination
            TabView(selection: $currentPage) {
                ForEach(0..<tutorialSteps.count, id: \.self) { index in
                    VStack(spacing: 16) {
                        // Image du tutoriel avec marges et style amélioré (plus large)
                        Image(tutorialSteps[index].imageName)
                            .resizable()
                            .aspectRatio(contentMode: .fill)
                            .frame(width: 350, height: 220)
                            .clipped()
                            .cornerRadius(20)
                            .shadow(color: .black.opacity(0.15), radius: 12, x: 0, y: 6)
                            .padding(.horizontal, 15)
                    }
                    .background(Color.white)
                    .tag(index)
                }
            }
            .background(Color.white)
            .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
            .frame(height: 250)
            
            // Indicateurs de page personnalisés
            HStack(spacing: 8) {
                ForEach(0..<tutorialSteps.count, id: \.self) { index in
                    Circle()
                        .fill(index == currentPage ? Color(hex: "#FD267A") : Color.gray.opacity(0.3))
                        .frame(width: 8, height: 8)
                        .animation(.easeInOut(duration: 0.3), value: currentPage)
                }
            }
            .padding(.top, 16)
            .padding(.bottom, 24)
            
            // Bouton de fermeture
            Button(action: {
                presentationMode.wrappedValue.dismiss()
            }) {
                Text("Compris !")
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 50)
                    .background(Color(hex: "#FD267A"))
                    .cornerRadius(25)
            }
            .padding(.horizontal, 20)
            .padding(.bottom, 20)
        }
        .background(Color.white)
        .cornerRadius(20)
    }
}

#Preview {
    WidgetTutorialView()
} 

File: Views/Tutorial/WidgetTutorialView.kt
@file:Suppress("unused")

package com.yourapp.ui.tutorial

import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.Image
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yourapp.R

data class TutorialStep(
    @StringRes val titleRes: Int,
    @DrawableRes val imageRes: Int
)

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun WidgetTutorialView(
    onClose: () -> Unit,
    steps: List<TutorialStep> = listOf(
        TutorialStep(R.string.widget_step_1, R.drawable.image_a),
        TutorialStep(R.string.widget_step_2, R.drawable.image_b),
        TutorialStep(R.string.widget_step_3, R.drawable.image_c),
        TutorialStep(R.string.widget_step_4, R.drawable.image_d),
    )
) {
    Surface(
        shape = RoundedCornerShape(20.dp),
        color = Color.White
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color.White)
        ) {
            // Handle (sheet grabber)
            Box(
                modifier = Modifier
                    .fillMaxWidth(),
                contentAlignment = Alignment.Center
            ) {
                Box(
                    modifier = Modifier
                        .padding(top = 12.dp, bottom = 8.dp)
                        .size(width = 40.dp, height = 5.dp)
                        .clip(RoundedCornerShape(2.5.dp))
                        .background(Color.Gray.copy(alpha = 0.3f))
                )
            }

            val pagerState = rememberPagerState(initialPage = 0, pageCount = { steps.size })

            // Titre dynamique selon la page courante
            Text(
                text = stringResource(id = steps[pagerState.currentPage].titleRes),
                modifier = Modifier
                    .padding(horizontal = 16.dp)
                    .padding(top = 20.dp, bottom = 20.dp)
                    .fillMaxWidth(),
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black,
                textAlign = TextAlign.Center
            )

            // Pager (équivalent TabView + PageTabViewStyle)
            HorizontalPager(
                state = pagerState,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(250.dp)
                    .background(Color.White)
            ) { page ->
                Column(
                    modifier = Modifier
                        .fillMaxSize(),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.Center
                ) {
                    // Image tutoriel (aspect fill + coins arrondis + ombre légère)
                    Card(
                        shape = RoundedCornerShape(20.dp),
                        elevation = CardDefaults.cardElevation(defaultElevation = 12.dp),
                        modifier = Modifier
                            .width(350.dp)
                            .height(220.dp)
                            .padding(horizontal = 15.dp)
                    ) {
                        Image(
                            painter = painterResource(id = steps[page].imageRes),
                            contentDescription = null,
                            contentScale = ContentScale.Crop,
                            modifier = Modifier.fillMaxSize()
                        )
                    }
                }
            }

            // Indicateurs de page
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = 16.dp, bottom = 24.dp),
                horizontalArrangement = Arrangement.Center
            ) {
                repeat(steps.size) { index ->
                    val selected = pagerState.currentPage == index
                    Box(
                        modifier = Modifier
                            .padding(horizontal = 4.dp)
                            .size(8.dp)
                            .clip(CircleShape)
                            .background(
                                if (selected) Color(0xFFFD267A)
                                else Color.Gray.copy(alpha = 0.3f)
                            )
                    )
                }
            }

            // Bouton "Compris"
            Button(
                onClick = onClose,
                shape = RoundedCornerShape(25.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFFD267A)),
                modifier = Modifier
                    .padding(horizontal = 20.dp, vertical = 20.dp)
                    .fillMaxWidth()
                    .height(50.dp)
            ) {
                Text(
                    text = stringResource(id = R.string.understood),
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White
                )
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun WidgetTutorialViewPreview() {
    WidgetTutorialView(onClose = {})
}
